src/components/AssetSearch/AssetList.tsx

```diff
@@ -23,17 +24,69 @@ type ItemData<T> = {
   handleClick: T
 }
 
+const sortAssetsByAccountAndMarketCap = (
+  assets: Asset[],
+  enrichedData: Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }>,
+): Asset[] => {
+  return sortBy(assets, [
+    asset => {
+      return Number(enrichedData[asset.assetId].fiatAmount)
+    },
+    asset => {
+      return Number(enrichedData[asset.assetId].marketCap)
+    },
+  ]).reverse()
+}
+
+/**
+ * This function create a data object(cryptoAmount, fiatAmount, marketCap) from the Asset, portfolioAcountRows and marketData for each asset for facilitate sorting.
+ * @param assets
+ * @param portfolioAcountRows
+ * @param marketData
+ * @returns Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }>
+ */
+export const createEnrichData = (
+  assets: Asset[],
+  portfolioAcountRows: AccountRowData[],
+  marketData: {
+    [x: string]: MarketData | undefined
+  },
+): Record<string, { fiatAmount: string; cryptoAmount: string; marketCap: string }> => {
+  const result = {} as Record<
+    string,
+    { fiatAmount: string; cryptoAmount: string; marketCap: string }
+  >
+  assets.forEach(asset => {
+    const fiatAmount = portfolioAcountRows.find(
+      portfolioAccountRow => portfolioAccountRow.assetId === asset.assetId,
+    )?.fiatAmount
+
+    const cryptoAmount = portfolioAcountRows.find(
+      portfolioAccountRow => portfolioAccountRow.assetId === asset.assetId,
+    )?.cryptoAmount
+
+    const assetMarketData = marketData[asset.assetId]
+
+    result[asset.assetId] = {
+      fiatAmount: bnOrZero(fiatAmount).toString(),
+      cryptoAmount: bnOrZero(cryptoAmount).toString(),
+      marketCap: bnOrZero(assetMarketData?.marketCap).toString(),
+    }
```

Comment:
These are already strings, we don't need to serialize to BN / deserialize to string what's already a string.
If the goal is to handle undefined values, we can use the nullish coercing operator and have a `?? ''` expression

Reply:
Ive seen it done both ways in our codebase. doesnt make a huge difference, imo its fine either way

Reply:
Ok in that case I will leave this line of code as it is now if you don't see any problem of course.